tests/serverpod_test_server/config/generator.yaml

```diff
@@ -6,4 +6,11 @@ modules:
   serverpod_test_module:
     nickname: module
   serverpod_auth:
-    nickname: auth
\ No newline at end of file
+    nickname: auth
+
+extraConstructors:
+  - project:src/custom_classes.dart:CustomClass
+  - project:src/custom_classes.dart:CustomClass?
```

Comment:
So should I register the nullable version for each extraConstructor by default? It would of course be possible.

I decided to name them differently on purpose.
- `extraConstructors` are those specified in the `generator.yaml`. They can be generated automatically, but the system can't detect them.
- `customConstructors` are specified by the user in the dart code. They are not generated by `serverpod generate`. They are for advanced users who want to add polymorphism for example.

Reply:
I think in this case, less may be more to keep the interface as clean as possible. The extraConstructors and extraClassNames have very similar information and their relation doesn't feel clear.

It may be smarter to just provide one way to provide extra classes (that is compatible with Freezed). If you do the serialization manually, it's up to you to make sure that you class conforms.

I suggest that we assume that any custom class have a constructor called `CustomClass.fromJson(dynamic)`. This way a custom class can be added as such:

```yaml
extraClasses:
  - CustomClass: project:src/custom_class.dart
```

Reply:
Like this? 7af9c728

Reply:
No, your syntax is different to what I described. Also, there should be no need to have a nullable constructor. This is confusing as constructors generally cannot return a null object.